src/draw/lv_draw.c

```diff
@@ -285,11 +303,16 @@ void lv_draw_dispatch_request(void)
 #endif
 }
 
+uint32_t lv_draw_get_unit_count(void)
+{
+    return _draw_info.unit_cnt;
+}
+
 lv_draw_task_t * lv_draw_get_next_available_task(lv_layer_t * layer, lv_draw_task_t * t_prev, uint8_t draw_unit_id)
 {
     LV_PROFILER_BEGIN;
     /*If the first task is screen sized, there cannot be independent areas*/
-    if(layer->draw_task_head) {
+    if(_draw_info.unit_cnt > 1 && layer->draw_task_head) {
```

Comment:
As the other `if(_draw_info.unit_cnt > 1)` is removed, it should be removed too.

Reply:
I believe we can leave it, this also saves some time.
Fill screen is the most consuming task. As all the tasks are handled by one single unit, I see no reason to wait for.

Reply:
With the current code if there is only 1 draw unit but 100 draw tasks are queued this `if` won't run. and all the 100 draw tasks will be checked below.  

We should return in all cases if a screen-sized area is queued. Even if I'm mistaken it's too risky to add such a change on the day of the release.

Reply:
Ok, If we revert also this change then basically all the performance code is gone :)
But we can keep them locally and create another PR for LVGL 9.3.

The performance was relying on those two ifs - first one already reverted due to SW tests failures.

Reply:
Ah, I see the problem. Let's keep it local if possible. I'm afraid that it will negatively affect pure SW rendering performance but don't have time to test it correctly now.

Reply:
Ok, I reverted all the dependency tasks and let them as before, done a separate commit to evidence them: 938e854266cb6c0367bd2e3130fa9ae697e98e30

We need to address it on the LVGL 9.3 as the difference on CPU usage is evident without the fix:
![image](https://github.com/user-attachments/assets/9ad59e3e-56f8-4ec3-afc0-2d9c97308e8e)

Reply:
I can't see any difference. It seems the `if`s aren't doing much difference.

Reply:
Look at the CPU usage.

Reply:
Okay, see now. So that `if` really helps lowering the performance.

Reply:
Both `if`s from commit https://github.com/lvgl/lvgl/commit/938e854266cb6c0367bd2e3130fa9ae697e98e30 are lowering the CPU usage, so would like to address them in LVGl 9.3.
Thanks!